tests/Microsoft.Identity.Web.Test/TokenAcquisitionAuthorityTests.cs

```diff
@@ -539,5 +539,358 @@ public async Task GetOrBuildManagedIdentity_TestConcurrencyAsync(string? clientI
+            Assert.False(string.IsNullOrEmpty(capturingHandler.CapturedClientAssertion));
+            var firstPayload = DecodeJwtPayload(capturingHandler.CapturedClientAssertion!);
+            Assert.Contains("v1", firstPayload, StringComparison.Ordinal);
+            // Attempt to change claims (should not affect cached app)
+            var newClaims = new Dictionary<string, string> { { "c1", "v2" }, { "c2", "vNew" } };
+            var newOptions = new TokenAcquisitionOptions
+            {
+                ExtraParameters = new Dictionary<string, object> { { "IDWEB_CLIENT_CLAIMS", newClaims } }
+            };
+            // Call GetOrBuild again with new claims
+            var app2 = await _tokenAcquisition.GetOrBuildConfidentialClientApplicationAsync(mergedOptions, newOptions);
+            // Same instance expected
+            Assert.Same(app2, await _tokenAcquisition.GetOrBuildConfidentialClientApplicationAsync(mergedOptions, null));
+            capturingHandler.ResetCapture();
+            var forceRefresh = new TokenAcquisitionOptions { ForceRefresh = true }; // network call
+            var second = await _tokenAcquisition.GetAuthenticationResultForAppAsync("https://graph.microsoft.com/.default", OpenIdConnectDefaults.AuthenticationScheme, tenantId, forceRefresh);
+            Assert.False(string.IsNullOrEmpty(capturingHandler.CapturedClientAssertion));
+            var secondPayload = DecodeJwtPayload(capturingHandler.CapturedClientAssertion!);
+            // Validate old value still present and new ones absent
+            Assert.Contains("v1", secondPayload, StringComparison.Ordinal);
+            Assert.DoesNotContain("v2", secondPayload, StringComparison.Ordinal);
+            Assert.DoesNotContain("vNew", secondPayload, StringComparison.Ordinal);
+        }
+
+        private static string DecodeJwtPayload(string jwt)
+        {
+            var parts = jwt.Split('.');
+            Assert.True(parts.Length >= 2, "JWT format invalid");
+            string payload = parts[1];
+            // Base64Url decode
+            string padded = payload.Replace('-', '+').Replace('_', '/');
+            switch (padded.Length % 4)
+            {
+                case 2: padded += "=="; break;
+                case 3: padded += "="; break;
+            }
+            var bytes = Convert.FromBase64String(padded);
+            return System.Text.Encoding.UTF8.GetString(bytes);
+        }
+
+        private class CapturingMsalHttpClientFactory : IMsalHttpClientFactory
+        {
+            private readonly HttpClient _httpClient;
+            public CapturingMsalHttpClientFactory(HttpClient httpClient) => _httpClient = httpClient;
+            public HttpClient GetHttpClient() => _httpClient;
+        }
+
+        private class CapturingHandler : HttpMessageHandler
```

Comment:
we already have a mocking framework to capture MSAL traffic and reply with test responses. i can update the tests to use them
see: https://github.com/AzureAD/microsoft-identity-web/blob/master/tests/Microsoft.Identity.Web.Test.Common/Mocks/MockHttpClientFactory.cs